### PR TITLE
Improved readme and function extension specification syntax improvements

### PR DESCRIPTION
**Pull request by acrusage-iaik**
Improved readme along with some breaking changes to improve definition and testing of function extensions.

### CODE REVIEW

**Comment by acrusage** [comment]
I'm unsure about the following though:
Global variables for the function extension definition have been created in the process. 
They were singletons before, but this is no longer possible as the functions are now instances. 
Are global variables fine like this, or what would be better practice?

**Comment by JesusMcCloud** [comment]
> I'm unsure about the following though: Global variables for the function extension definition have been created in the process. They were singletons before, but this is no longer possible as the functions are now instances. Are global variables fine like this, or what would be better practice?

You mean as part of the object, or did I overlook something?

**Review by JesusMcCloud — CHANGES_REQUESTED** [review]
Readme looks good!

One minor thing:
Use the functional Napier interface (e.g `Napier.e{}` (curly braces!)), as it completely omits all logging invocations for disabled log levels.

File: jsonpath/src/commonMain/kotlin/at/asitplus/jsonpath/core/functionExtensions/countFunctionExtension.kt [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+import kotlinx.serialization.ExperimentalSerializationApi
+import kotlinx.serialization.json.JsonPrimitive
+
+/**
+ * specification: https://datatracker.ietf.org/doc/rfc9535/
+ * date: 2024-02
+ * section: 2.4.5.  count() Function Extension
+ */
+@OptIn(ExperimentalSerializationApi::class)
+internal val countFunctionExtension by lazy {
```
Comment (acrusage-iaik) [inline jsonpath/src/commonMain/kotlin/at/asitplus/jsonpath/core/functionExtensions/countFunctionExtension.kt:15]: @JesusMcCloud Regarding https://github.com/a-sit-plus/jsonpath/pull/9#issuecomment-2100896623
Is there a better way to define this extension without making it a global variable?

**Comment by acrusage-iaik** [comment]
Fixed in commit a918d6b1d0574ae9c69bfecbc167b72632b0ed4d

**Comment by acrusage-iaik** [comment]
> Readme looks good!
> 
> One minor thing: Use the functional Napier interface (e.g `Napier.e{}` (curly braces!)), as it completely omits all logging invocations for disabled log levels.

Fixed in commit a918d6b1d0574ae9c69bfecbc167b72632b0ed4d

**Review by JesusMcCloud — APPROVED** [review]

**Pull request merged**